pallets/gear/src/lib.rs

```diff
@@ -456,6 +485,10 @@ pub mod pallet {
         GearRunAlreadyInBlock,
         /// The program rent logic is disabled.
         ProgramRentDisabled,
+        /// User messages to this destination are not allowed.
+        ///
+        /// Occurs if a user attempts to send a message to a built-in actor.
+        IllegalDestination,
```

Comment:
why is it illegal to user to send messages to built-in actor?

Reply:
Maybe they could, but what is the use case? Users can directly call extrinsics from other runtime pallets.
Allowing users to send messages to built-in actor would complicate implementation - we'd need to be able to prepare a user reply message on the built-in actor side.
Let's think about it

Reply:
Ok, re-enabled user messages to the built-in actor.
Indeed, it is a legit flow. However, sending messages from a user to the built-in actor doesn't make sense economically, therefore we shouldn't expect anyone wanting to do this... often.

Reply:
Those users can be bridges or some other external actors, not actual users, so it should be fine